### DISCUSSION TITLE
How to add frontmatter to files not created by obsidian.nvim

### DISCUSSION THREAD

**Discussion by sotte** [epwalsh/obsidian.nvim#335] [Q&A] [answered]
I 'm migrating my org notes to markdown (and obsidian.nvim). Is there a way to toggle creating the frontmatter for the new files?

I copied the converted files into my fault, but obsidian.nvim does not add a frontmatter. It only adds frontmatter for new files that I created with obsidian.nvim. 

Also: any suggestions how to script this? I have a few thousand notes and don't want to do it manually.

**Comment by epwalsh** [comment, marked as answer]
Hey @sotte should be scriptable. Create a Lua file (`tmp.lua`) with something like this:

```lua
local scan = require("plenary.scandir")
local Note = require("obsidian.note")
local client = require("obsidian").get_client()

local paths = scan.scan_dir(tostring(client.dir), { search_pattern = ".*.md$" })
for _, path in ipairs(paths) do
	print(string.format("Updating note at %s...", path))
	local note = Note.from_file(path, client.dir)
	note:save()
end
```

Then from a buffer in your vault (obsidian.nvim needs to be loaded), run `:luafile tmp.lua`.

**Reply by epwalsh** [reply]
`note:save()` will write the frontmatter and leave the rest of the contents alone.

**Reply by epwalsh** [reply]
Once #336 merges you could simplify that script to this:

```lua
local client = require("obsidian").get_client()

client:apply_async(function(note)
  print(string.format("Updating note %s...", note.path))
  note:save()
end)
```

**Reply by sotte** [reply]
I'll post an update once I'm done with the migration. But this looks good. Thanks!

**Reply by sotte** [reply]
Worked like a charm. Thanks @epwalsh!

**Reply by theherk** [reply]
Another person needing this after a migration from org. This latter script worked like an absolute champ. Thank @epwalsh.
